src/api/lib/backend/api/cloud.rb

```diff
@@ -15,6 +15,12 @@ def self.upload(user, params, target_name = 'ec2')
       def self.status(user)
         get(['/cloudupload'], params: { name: user.upload_jobs.pluck(:job_id) }, expand: [:name])
       end
+
+      # Returns the log file of the cloud upload job
+      # @return [String]
+      def self.log(id)
+        get(["/cloudupload/#{id}/_log"])
```

Comment:
Please, don't interpolate strings directly in the endpoint, for that better add it as a placeholder and add the variable in the array like this:

```ruby
get(["/cloudupload/:id/_log", id])
```
That way it gets escaped properly to be url compliant (I know that it will be an integer, but in general is the best aproach).

BTW, for the endpoints in the other methods you don't need to pass an array (like we do at https://github.com/openSUSE/open-build-service/blob/master/src/api/lib/backend/api/build_results/worker.rb#L11) because it don't have any placeholder to replace, so in lines 10 and 16 of this file just do this:

```ruby
# line 10
post('/cloudupload', params: params.merge(user: user.login, target: target_name), data: data)

# line 16
get('/cloudupload', params: { name: user.upload_jobs.pluck(:job_id) }, expand: [:name])
```

Reply:
Yes we have documentation, is generated using Yard (do you remember the discussion about it?) check this out at src/api/lib/backend/doc/index.html 

For knowing how to use this see the ConnectionHelper module